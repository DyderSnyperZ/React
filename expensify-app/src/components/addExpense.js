import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { addState } from '../actions/expenses';
import ExpenseForm from './expenseForm';

const AddExpensePage = (props) => (
    <div>
        <ExpenseForm onSubmit={expense => { // récupère les donées expense du formulaire
            props.dispatch(addState(expense));
            props.history.push('/'); // redirect to route
        }}/>
    </div>
);

export default connect()(AddExpensePage);