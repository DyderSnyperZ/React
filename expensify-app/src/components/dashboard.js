import React from 'react';
import ExpenseListe from './expenseList';
import ExpenseListFilter from './expenseListFilter';
import SortBy from './sortFilter';

const DashboardPage = () => (
    <div>
        <ExpenseListFilter />
        <ExpenseListe />
    </div>
);

export default DashboardPage;