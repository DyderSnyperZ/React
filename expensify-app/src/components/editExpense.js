import React from 'react';
import ExpenseForm from './expenseForm';
import { connect } from 'react-redux';
import { editState, removeState } from '../actions/expenses';

const EditExpensePage = (props) => (
    <div>
        <ExpenseForm
            expense={props.expense}
            onSubmit={(expense) => {
                props.dispatch(editState(props.expense.id, expense )); //edit expense
                props.history.push('/'); // redirect to route
            }} />
            <button onClick={() => {
                props.dispatch(removeState(({id: props.expense.id})));
                props.history.push('/'); // redirect to route
        }}
        >Remove
</button>
    </div>
);

const MapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find(element => element.id === props.match.params.id) // retourne expense par id qu'on souhaite
    }
}

export default connect(MapStateToProps)(EditExpensePage);