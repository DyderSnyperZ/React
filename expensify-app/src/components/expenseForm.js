import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

export default class ExpenseForm extends React.Component {
    constructor(props){
        super(props); // récupère ce qui est passé dans le parent
        this.state= { // set données si existe dans parent
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.note : '',
            price: props.expense ? (props.expense.price / 100).toString() : '',
            calendarFocused: false,
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            error: ''
        }
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    };
    onPriceChange = (e) => {
        const price = e.target.value;
        if (price && price.match(/^\d{1,}(\.\d{0,2})?$/)) // récupère uniquement les chiffre et décimal si existe
            this.setState(() => ({ price }));
    };
    onDateChange = (createdAt) => {
        if (createdAt)
            this.setState(() => ({ createdAt }));
    };
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };
    handleSubmitForm = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.price) {
            this.setState(() => ({ error: 'Please provide description and amount' }));
        } else {
            this.props.onSubmit({ // fonction du parent ou on lui envoie les nouvelles données
                description: this.state.description,
                price: parseFloat(this.state.price, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note  
            });
        }
    };
    render() {
        return (
            <div>
            {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.handleSubmitForm}>
                    <input type="text"
                        name="description"
                        placeholder="description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange} />
                    <input type="text"
                        name="price"
                        placeholder="price"
                        value={this.state.price}
                        onChange={this.onPriceChange} />
                    <textarea
                        placeholder="Add note here"
                        onChange={this.onNoteChange}
                        value={this.state.note}
                    ></textarea>
                    <SingleDatePicker //set SingleDatePicker https://github.com/airbnb/react-dates
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false} />
                    <button>Test</button>
                </form>
            </div>
        )
    }
}