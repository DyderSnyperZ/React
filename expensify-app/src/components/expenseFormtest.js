import React from 'react';
import { connect } from 'react-redux';
import { addState } from '../actions/expenses';

const ExpenseForm = ({ dispatch }) => (
    <div>
        <form name="expenseForm" onSubmit={e => { e.preventDefault(); handleSubmitForm(e, dispatch) }}>
            <input type="text" name="square"></input>
            <input type="text" name="description"></input>
            <input type="text" name="price"></input>
            <input type="text" name="createdAt"></input>
            <button>Test</button>
        </form>
    </div>
)

const handleSubmitForm = (e, dispatch) => {
    const jsonData = {
        square: e.target.square.value,
        description: e.target.description.value,
        price: e.target.price.value,
        createdAt: e.target.createdAt.value
    };

    dispatch(addState(jsonData));
};

export default connect()(ExpenseForm);