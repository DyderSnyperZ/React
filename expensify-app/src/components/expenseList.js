import { connect } from 'react-redux';
import React from 'react';
import ExpenseListeItem from './expenseListItem';
import filteredExpenses from '../selectors/expenses';

export const ExpenseListe = (props) => ( //component a render
    <div>
    <h2>Expenses List</h2>
        {props.expenses.length === 0 ? (
            <p>No expenses</p>
        ) : (
            <u1>
                {props.expenses.map(expense => <ExpenseListeItem key={expense.id}{...expense} />)}
            </u1>
        )
        }
    </div>
);

const mapStateToProps = (state) => { // maps les valeurs de state au variables du template
    return {
        expenses: filteredExpenses(state.expenses, state.filter)
    }
};

export default connect(mapStateToProps)(ExpenseListe); //connect les variables redux store avec les variables du template



