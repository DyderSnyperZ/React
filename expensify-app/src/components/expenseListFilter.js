import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter } from '../actions/filters';
import { sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilter extends React.Component {
    state = {
        calendarFocused:null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    }
    render() {
        return (
            <div>
                <input type="text" value={this.props.filter.text} onChange={(e) => {
                    this.props.dispatch(setTextFilter(e.target.value)); // set dans le store les data avec le tri qu'on veux
                }}></input>
                <select value={this.props.filter.sortBy} onChange={(e) => {
                    switch (e.target.value) {
                        case 'date':
                            this.props.dispatch(sortByDate());
                        case 'amount':
                            this.props.dispatch(sortByAmount());
                    }
                }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                startDate={this.props.filter.startDate}
                endDate={this.props.filter.endDate}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
            />
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    return {
        filter: state.filter
    }
};

export default connect(mapStateToProps)(ExpenseListFilter);