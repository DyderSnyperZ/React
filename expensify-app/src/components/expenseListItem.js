import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { removeState } from '../actions/expenses';

export const ExpenseListeItem = ({ dispatch, id, description, price, createdAt }) => ( //component a render
        <li >
                <Link to={`/edit/${id}`}>Description : {description}</Link>
         - Price : {price} - createdAt : {createdAt}
        </li>
);

export default connect()(ExpenseListeItem);


