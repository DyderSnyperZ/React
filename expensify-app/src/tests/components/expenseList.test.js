import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseListe } from '../../components/expenseList';
import tabExpenses from '../fixtures/expenses';

test('expense list with passed data', () => {
    const wrapper = shallow(<ExpenseListe expenses={tabExpenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('expense list with no passed data', () => {
    const wrapper = shallow(<ExpenseListe expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});

