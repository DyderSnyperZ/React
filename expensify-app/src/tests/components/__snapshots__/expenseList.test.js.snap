// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expense list with no passed data 1`] = `
<div>
  <h2>
    Expenses List
  </h2>
  <p>
    No expenses
  </p>
</div>
`;

exports[`expense list with passed data 1`] = `
<div>
  <h2>
    Expenses List
  </h2>
  <u1>
    <Connect(ExpenseListeItem)
      createdAt={"1970-01-01T00:00:00.000Z"}
      description="Water Bill"
      id="1"
      key="1"
      price="2500"
      square=""
    />
    <Connect(ExpenseListeItem)
      createdAt={-345600000}
      description="Gaz Bisssll"
      id="2"
      key="2"
      price="25000"
      square=""
    />
    <Connect(ExpenseListeItem)
      createdAt={345600000}
      description="Train Bill"
      id="3"
      key="3"
      price="20"
      square=""
    />
  </u1>
</div>
`;
